confidence/models.py

```diff
@@ -194,7 +199,13 @@ def get(self,
                 return as_type(value)
             elif isinstance(value, Mapping):
                 # wrap value in a Configuration
-                return self._wrap(value)
+                value = self._wrap(value)
+                if self._secrets and self._secrets.matches(value):
```

Comment:
NB: this gets called every step along the `config.deeply.nested.keys.foobar` way; `matches()` needs to be cheap :grimacing: